measurements-to-gelf.py

```diff
@@ -0,0 +1,96 @@
+#!/usr/bin/python
+
+#
+# Prototype of a RIPE Atlas to GELF parser by awlx 
+#
+# Usage: python measurements-to-gelf.py <id> <min>
+#
+
+# Imports
+from gelfclient import UdpClient
+from urllib2 import urlopen
+import syslog
+import json
+import time
+import sys
+import requests
+
+# Configuration
+gelf_server = 'localhost'
+gelf_port = 5555
+api_key = '<api_key>'
+
+# Parse Args
+if len(sys.argv) == 1:
+    print "# Prototype of a RIPE Atlas to GELF parser"
+    print "#"
+    print "# Usage: python measurements-to-gelf.py <measurement id> <timeframe in min>"
+    print "#"
+    print "# Example: python measurements-to-gelf.py 12323 5"
+    print "#"
+    print "# Please define Server and Port inside the script and you need an API Key for Geolocation from https://geocoder.opencagedata.com/pricing"
+    exit(1) 
+
+measurement_id = str(sys.argv[1])
+timeframe_secs = int(sys.argv[2]) * 60
+
+
+
+def getplace(lat, lon):
+    url = "http://api.opencagedata.com/geocode/v1/json?q="
+    url += "%s+%s&key=%s" % (lat, lon, api_key)
+    v = urlopen(url).read()
+    j = json.loads(v)
+    components = j['results'][0]['components']
+    country = town = None
+    try:
+        country = components['country']
+        state = components['state']
+    except Exception as e:
+        country = "N/A"
+        state = "N/A"
+    return state, country
+
+
+currenttime = int(time.time())
+timeframe = currenttime - timeframe_secs
+
+measurement_url = "https://atlas.ripe.net/api/v2/measurements/%s/results?start=%s&stop=%s&format=json" % (measurement_id, str(timeframe), str(currenttime))
```

Comment:
This line exceeds 80 characters, but it also uses an antiquated way of string formatting: the `%` operator.  While this is still supported in modern versions of Python, it's been marked as deprecated and we're all supposed to use `.format()` now (which is supported in 2.6+ if not even earlier).

Additionally, it's generally a bad idea to manually assemble URL query strings like this as you run the risk of introducing characters that aren't encoded properly (like spaces, slashes, etc.)  Instead, Python has a handy library for that.  In Python2 it's `urllib.urlencode` and in Python3 it's `urllib.parse.urlencode`, so you would typically do it like this:

```python
# At the top of the file
try:
    from urllib.parse import urlencode
except ImportError:
    from urllib import urlencode

# Further down where you need it
measurement_url = "https://atlas.ripe.net/api/v2/measurements/{}/results?{}".format(
    measurement_id,
    urlencode({
        "start": time_frame,
        "stop": current_time,
        "format": "json"
    })
)
```